sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/frequency/SketchFrequencies.java

```
@@ -0,0 +1,357 @@
+import java.util.Iterator;
+
+import org.apache.beam.sdk.coders.ByteArrayCoder;
+import org.apache.beam.sdk.coders.CannotProvideCoderException;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.coders.CoderException;
+import org.apache.beam.sdk.coders.CoderRegistry;
+import org.apache.beam.sdk.coders.CustomCoder;
+import org.apache.beam.sdk.transforms.Combine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@code PTransform}s that records an estimation of the frequency of each element in a
+ * {@code PCollection}, or the occurrences of values associated with each key in a
+ * {@code PCollection} of {@code KV}s.
+ *
+ * <p>This class uses the Count-min Sketch structure. The papers and other useful information
+ * about it is available on this website : <a>https://sites.google.com/site/countminsketch/</a>
+ * <br>The implementation comes from Apache Spark :
+ * <a>https://github.com/apache/spark/tree/master/common/sketch</a>
+ */
+public class SketchFrequencies {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SketchFrequencies.class);
+
+  // do not instantiate
+  private SketchFrequencies() {
+  }
+
+  /**
+   * A {@code PTransform} that takes an input {@code PCollection<String>} and returns a
+   * {@code PCollection<CountMinSketch>} whose contents is a Count-min sketch that allows to query
+   * the number of hits for a specific element in the input {@code PCollection}.
+   *
+   * <p>The {@code seed} parameters will be used to randomly generate different hash functions.
+   * Thus, the result can be different for the same stream in different seeds are used.
+   * The {@code seed} parameter will be used to generate a and b for each hash function.
+   * <br>The Count-min sketch size is constant through the process so the memory use is fixed.
+   * However, the dimensions are directly linked to the accuracy.
+   * <br>By default, the relative error is set to 1% with 1% probability that the estimation
```

Since this returns a `Combine.Globally`, it's not possible to change this on the result of this transform, so "by default" is somewhat misleading.

I think error bounds are something that the user should **always** configure explicitly and there does not exist a reasonable default value - it is entirely application-specific. And the current function signature does not allow configuring them.

There are a few ways out of this:
1) Instead of returning a `Combine.Globally`, create a new wrapper transform for this, with builder methods for relevant parameters. It will be rather boilerplate-y.
2) Just remove this function and expose only `CountMinSketchFn`, and have examples in javadoc about how to use it in a pipeline, mentioning various scenarios: `Combine.Globally`, `Combine.PerKey`, state cells etc. (not necessary to give code examples for each)
3) Add more parameters to the current function (after removing `seed` :) )

I prefer 2 - in particular, because we can always add convenience wrappers later if it turns out that life is too hard without them - but I could be convinced otherwise.